go/vt/vtgate/resilient_srv_topo_server_test.go

```diff
@@ -298,21 +313,41 @@ func TestCacheWithErrors(t *testing.T) {
 	rsts := NewResilientSrvTopoServer(topo.Server{Impl: ft}, "TestCacheWithErrors")
 
 	// ask for the known keyspace, that populates the cache
-	_, err := rsts.GetSrvKeyspace(context.Background(), "", "test_ks")
+	_, err := rsts.GetSrvShard(context.Background(), "", "test_ks", "shard_0")
 	if err != nil {
-		t.Fatalf("GetSrvKeyspace got unexpected error: %v", err)
+		t.Fatalf("GetSrvShard got unexpected error: %v", err)
 	}
 
 	// now make the topo server fail, and ask again, should get cached
 	// value, not even ask underlying guy
 	ft.keyspace = "another_test_ks"
-	_, err = rsts.GetSrvKeyspace(context.Background(), "", "test_ks")
+	_, err = rsts.GetSrvShard(context.Background(), "", "test_ks", "shard_0")
 	if err != nil {
-		t.Fatalf("GetSrvKeyspace got unexpected error: %v", err)
+		t.Fatalf("GetSrvShard got unexpected error: %v", err)
 	}
 
 	// now reduce TTL to nothing, so we won't use cache, and ask again
 	rsts.cacheTTL = 0
+	_, err = rsts.GetSrvShard(context.Background(), "", "test_ks", "shard_0")
+	if err != nil {
+		t.Fatalf("GetSrvShard got unexpected error: %v", err)
+	}
+}
+
+// TestSrvKeyspaceCacheWithErrors will test we properly return cached errors for GetSrvKeyspace.
+func TestSrvKeyspaceCacheWithErrors(t *testing.T) {
```

Comment:
We need a test that writes a new value of SrvKeyspace on fakeTopo.notifications, and makes sure subsequent GetSrvKeyspace get the new value. That is the core unit test we need. Watch out for flaky unit tests as notification is asynchronous, probably need a loop with a timeout until GetSrvKeyspace gets the value.